feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts

```diff
@@ -157,6 +214,13 @@ export class AsmMainUiComponent implements OnInit, OnDestroy {
     this.csAgentAuthService.authorizeCustomerSupportAgent(userId, password);
   }
 
+  private startSessionWithParameters(): void {
+    if (this.customerIdInURL && !this.emulated) {
```

Comment:
We will be accounting for many potential parameters in the very near future than just `customerId`. We should come up with a more robust solution to predicate this situation. For example:

```
getParametersFromUrl() {
  const ALLOWED_PARAMETERS = ['customerId', *future params*...]
  const queryString = this.location.path().split('?')[1];
  const params = ALLOWED_PARAMETERS.map(param => ({ [param]: new URLSearchParams(queryString).get(param)}))
  return params
}

...

startSessionWithParameters(this.getParametersFromUrl())
```

Reply:
Did similar work in asmComponentService, not sure we need an allowedList.